### ISSUE TITLE
Add context as parameter to all methods in Storager and ProjectStorager

### ISSUE THREAD

**Issue by furuholm** [Grafeas/Grafeas#86]
@skelterjohn suggested this in a comment regarding PR https://github.com/Grafeas/Grafeas/pull/81#discussion_r158976153

> These interface methods should have a context as the first parameter. With context, you go all in. Annoying to plumb through, but the idea is that perhaps the underlying .CreateProject() call also makes an RPC to write the data - it would want to time out on the parent context as well.

Sounds like good idea to me but I am making it its own issue as it is outside the scope of the PR.

**Comment by furuholm** [comment]
BTW: I can provide a PR for this once the list projects PR has been merged.

**Comment by vtsao** [comment]
Yeah, all the storage interface methods should be updated to have ctx as the first param. SGTM, feel free to send a PR.

**Comment by furuholm** [comment]
Looking at this again I'm leaning towards this being premature. Unless there is a concrete use case for this right now I prefer not making the change. Also: an eventual addition of an RPC to create projects would probably end up in the impl-layer rather than the storage layer anyway.

I suggest we close this. Thoughts @R2wenD2?
